pkg/controller/release/rolloutblock.go

```diff
@@ -0,0 +1,63 @@
+package release
+
+import (
+	"fmt"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/runtime"
+
+	shipper "github.com/bookingcom/shipper/pkg/apis/shipper/v1alpha1"
+	shippererrors "github.com/bookingcom/shipper/pkg/errors"
+	rolloutblockUtil "github.com/bookingcom/shipper/pkg/util/rolloutblock"
+	stringUtil "github.com/bookingcom/shipper/pkg/util/string"
+)
+
+func (s *Scheduler) shouldBlockRollout(rel *shipper.Release) (bool, string, error) {
+	relOverrideRB, ok := rel.Annotations[shipper.RolloutBlocksOverrideAnnotation]
+	if !ok {
+		relOverrideRB = ""
+	}
+
+	nsRBs, err := s.rolloutBlockLister.RolloutBlocks(rel.Namespace).List(labels.Everything())
+	if err != nil {
+		runtime.HandleError(fmt.Errorf("error syncing Release %q Because of namespace RolloutBlocks (will retry): %s", rel.Name, err))
+	}
+
+	gbRBs, err := s.rolloutBlockLister.RolloutBlocks(shipper.ShipperNamespace).List(labels.Everything())
+	if err != nil {
+		runtime.HandleError(fmt.Errorf("error syncing Release %q Because of global RolloutBlocks (will retry): %s", rel.Name, err))
```

Comment:
I suggest reflecting the error essentials here, like: `failed to list rollout block objects: %s`